Adds Orangecrab starter project (#38)
clash-lang/stack-templates: projects/orangecrab/README.md

# OrangeCrab Development Setup

> This sub-project contains some demo applications for targeting the [OrangeCrab FPGA r0.2.1](https://orangecrab-fpga.github.io/orangecrab-hardware/docs/r0.2.1) in combination with the OrangeCrab Pmod / Debug Expander board and an [Adafruit FT232H JTAG Debugger](https://learn.adafruit.com/adafruit-ft232h-breakout).

Reviewer: I suspect the notion of a _sub-project_ refers to where the code originates. It doesn't seem apt here.

* `make` is used to run `clash`, RTL synthesis, Place and Route, and for programming the FPGA.

### Cabal

In terms of Haskell / Clash package management, everything is fairly standard. Have a look at the [`orangecrab.cabal`](orangecrab.cabal) and the [`cabal.project`](../cabal.project) files for the particular details.

### Make

A straightforward [`makefile`](makefile) is used on top of `cabal` to handle any additional calls which are not covered by the cabal build system yet.

#### Configuration

The `makefile` includes the additional configuration file [`build.cfg`](build.cfg), which can be used to link all of the required external build tools. To this end, simply copy the file to `build.cfg.local` on your local machine and modify it towards your personal needs. Any changes made to the copy will be preferred in comparison to original `build.cfg`. Just leave out the overrides, if you like to stick with the defaults instead.

Some references on where to get all of the required external tooling are also given as part of the configuration file.

> The particular top entity to be built can be selected via overriding the `NAME` parameter in the `build.cfg.local`. Just set it to the name of the file in the `src` directory shall be built (without the `.hs` ending). See the `makefile` for the pre-configured default.

Reviewer: ```suggestion
The particular top entity to be built can be selected via overriding the `NAME` parameter in the `build.cfg.local`. Just set it to the name of the file in the `src` directory shall be built (without the `.hs` ending). See the `Makefile` for the pre-configured default.
```

#### Usage

> The `makefile` offers the follwing commands (to be invoked via `make CMD`) where each command depends on the output generated by all of the former ones according to the order of the list below. The `bitstream` command is the `make` default.

Reviewer: ```suggestion
The `Makefile` offers the follwing commands (to be invoked via `make TARGET`) where each tareget depends on the output generated by all of the former ones according to the order of the list below. The `bitstream` target is the `make` default.
```

> | CMD         | Description                                         |

Reviewer: ```suggestion
| TARGET      | Description                                         |
```
